src/transport/raw/TCP.cpp

```diff
@@ -359,126 +422,171 @@ CHIP_ERROR TCPBase::OnTcpReceive(Inet::TCPEndPoint * endPoint, System::PacketBuf
     return CHIP_NO_ERROR;
 }
 
-void TCPBase::OnConnectionComplete(Inet::TCPEndPoint * endPoint, CHIP_ERROR inetErr)
+void TCPBase::HandleTCPEndPointConnectComplete(Inet::TCPEndPoint * endPoint, CHIP_ERROR conErr)
 {
     CHIP_ERROR err          = CHIP_NO_ERROR;
     bool foundPendingPacket = false;
     TCPBase * tcp           = reinterpret_cast<TCPBase *>(endPoint->mAppState);
     Inet::IPAddress ipAddress;
+    char ipAddrStr[64];
     uint16_t port;
     Inet::InterfaceId interfaceId;
+    ActiveConnectionState * activeConnection = nullptr;
 
     endPoint->GetPeerInfo(&ipAddress, &port);
     endPoint->GetInterfaceId(&interfaceId);
     PeerAddress addr = PeerAddress::TCP(ipAddress, port, interfaceId);
+    ipAddress.ToString(ipAddrStr, sizeof(ipAddrStr));
```

Comment:
Does that method not have an overload taking a char array?

Reply:
Yes, it does. Initially I thought it didn't. Removed the sizeof.